src/modelgauge/pipeline.py

```diff
@@ -70,11 +70,11 @@ class PipelineSegment(ABC):
 
     default_timeout = 0.1
 
-    def __init__(self):
+    def __init__(self, queue_maxsize=0):
         super().__init__()
         self._work_done = Event()
         self._upstream: Optional[PipelineSegment] = None
-        self._queue: Queue = Queue()
+        self._queue: Queue = Queue(queue_maxsize)
```

Comment:
I'm curious what this is meant for

Reply:
Before we had unbounded queues. This meant that it would, e.g., read in all prompts and queue items for every prompt * sut * annotator. And then those would sit there until one of the workers was ready. That's fine for small numbers of things, but when it's a lot of things, you get a lot of unnecessary memory usage, and can also get other behaviors.

For example, I think some of our overloading Together enough to get errors was making SUT requests long before we were ready to annotate them. But now, since the queue has a limited size, it'll only do so many ahead of need.

Reply:
Oh got it, that makes sense. Thanks!